salt/utils/network.py

```diff
@@ -589,7 +589,10 @@ def _interfaces_ifconfig(out):
                     if not salt.utils.is_sunos():
                         ipv6scope = mmask6.group(3) or mmask6.group(4)
                         addr_obj['scope'] = ipv6scope.lower() if ipv6scope is not None else ipv6scope
-                if addr_obj['address'] != '::' and addr_obj['prefixlen'] != 0:  # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                if salt.utils.is_sunos():
+                    if addr_obj['address'] != '::' and addr_obj['prefixlen'] != 0:  # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                        data['inet6'].append(addr_obj)
+                else:
                     data['inet6'].append(addr_obj)
```

Comment:
What are we actually trying to do here. From what I can tell, we are supposed to append ``addr_obj`` if one of the following is true:

1. We are not on SunOS
2. We are on SunOS, and both ``addr_obj['address'] != '::'`` and ``addr_obj['prefixlen'] != 0``

If this is the case, the question is the same as the last PR you opened for this: why have two separate appends? The same can be done in a single ``if`` block:

```diff
diff --git a/salt/utils/network.py b/salt/utils/network.py
index 529850b..35638c6 100644
--- a/salt/utils/network.py
+++ b/salt/utils/network.py
@@ -589,7 +589,10 @@ def _interfaces_ifconfig(out):
                     if not salt.utils.is_sunos():
                         ipv6scope = mmask6.group(3) or mmask6.group(4)
                         addr_obj['scope'] = ipv6scope.lower() if ipv6scope is not None else ipv6scope
-                if addr_obj['address'] != '::' and addr_obj['prefixlen'] != 0:  # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                if not salt.utils.is_sunos() \
+                        or addr_obj['address'] != '::' \
+                        and addr_obj['prefixlen'] != 0:
                     data['inet6'].append(addr_obj)
         data['up'] = updown
         if iface in ret:
```

Python's lazy evaluation means that if we're not on SunOS, we never attempt to check the ``address`` or ``prefixlen``. The only way we get to that part of the ``if`` statement is if we're on SunOS.